src/tree/azure/CustomAzureResourceTreeDataProvider.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureResource } from 'api/docs/vscode-azureresources-api';
+import * as vscode from 'vscode';
+import { AzExtResourceType, ResourceModelBase } from '../../../api/src/index';
+import { AzureResourceProviderManager } from '../../api/ResourceProviderManagers';
+import { azureExtensions } from '../../azureExtensions';
+import { showHiddenTypesSettingKey } from '../../constants';
+import { ext } from '../../extensionVariables';
+import { settingUtils } from '../../utils/settingUtils';
+import { BranchDataItemCache } from '../BranchDataItemCache';
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+import { TreeItemStateStore } from '../TreeItemState';
+import { AzureResourceGroupingManager } from './AzureResourceGroupingManager';
+import { AzureResourceTreeDataProviderBase } from './AzureResourceTreeDataProviderBase';
+import { GroupingItem } from './GroupingItem';
+
+const supportedResourceTypes: AzExtResourceType[] =
+    azureExtensions
+        .map(e => e.resourceTypes)
+        .reduce((a, b) => a.concat(...b), []);
+
+export class CustomAzureResourceTreeDataProvider extends AzureResourceTreeDataProviderBase {
```

Comment:
nit: Are you going to change the name of this? It's pretty confusing as if what this is supposed to be for.

Reply:
Updated